track_oracle/core/kwiver_io_base.txx

```diff
@@ -30,10 +30,18 @@ typedef char (&no_io_type)[2];
 typedef char yes_io_type;
 
 struct proxy{ template <typename U> proxy(const U&); };
+# if defined(_MSC_VER)
+#  pragma warning( push )
+#  pragma warning( disable : 4172 ) /* returning address of local variable */
```

Comment:
I'm wondering if other compilers will generate similar warnings.  Regardless of whether the function is actually every called or not, is there a better way to implement it to avoid this warning in the first place?  I suppose one option is to make the dummy variable global.  Does `no_io_type` have to be a reference for this to work?  I haven't fully grasped what the meta-program is doing so I don't have any other suggestions off the top of my head.